rust/lance-encoding/src/encoder.rs

```diff
@@ -810,6 +810,19 @@ impl CompressionStrategy for CoreArrayEncodingStrategy {
         }
         if let DataBlock::VariableWidth(ref variable_width_data) = data {
             if variable_width_data.bits_per_offset == 32 {
+                let data_size = variable_width_data.get_stat(Stat::DataSize).expect(
+                    "VariableWidth DataBlock should have valid `Stat::DataSize` statistics",
+                );
+                let data_size = data_size.as_primitive::<UInt64Type>().value(0);
```

Comment:
Also could be helpful to have a `expect_single_state` / `get_single_stat` method.  Then you can just do:

```
let data_size = variable_width_data.expect_single_stat::<UInt64Type>(Stat::DataSize);
```

Reply:
thanks for the suggestion! I will create a separate PR for this.